From ad866a1ca3e7d60da888d25d27e46a8adb2ed36e Mon Sep 17 00:00:00 2001
From: Natalia <124304+nessita@users.noreply.github.com>
Date: Mon, 6 Jan 2025 15:51:45 -0300
Subject: [PATCH] [4.2.x] Fixed CVE-2024-56374 -- Mitigated potential DoS in
 IPv6 validation.

Thanks Saravana Kumar for the report, and Sarah Boyce and Mariusz
Felisiak for the reviews.

Co-authored-by: Natalia <124304+nessita@users.noreply.github.com>
---
 django/db/models/fields/__init__.py           |  6 +--
 django/forms/fields.py                        |  7 +++-
 django/utils/ipv6.py                          | 19 +++++++--
 docs/ref/forms/fields.txt                     | 13 +++++-
 docs/releases/4.2.18.txt                      | 12 ++++++
 .../field_tests/test_genericipaddressfield.py | 33 ++++++++++++++-
 tests/utils_tests/test_ipv6.py                | 40 +++++++++++++++++--
 7 files changed, 116 insertions(+), 14 deletions(-)

diff --git a/docs/ref/forms/fields.txt b/docs/ref/forms/fields.txt
index 1a7274e0d1ad..76b4587e21a6 100644
--- a/docs/ref/forms/fields.txt
+++ b/docs/ref/forms/fields.txt
@@ -742,6 +742,15 @@ For each field, we describe the default widget used if you don't specify
         ``192.0.2.1``. Default is disabled. Can only be used
         when ``protocol`` is set to ``'both'``.
 
+    .. attribute:: max_length
+
+        Defaults to 39, and behaves the same way as it does for
+        :class:`CharField`.
+
+    .. versionchanged:: 4.2.18
+
+        The default value for ``max_length`` was set to 39 characters.
+
 ``ImageField``
 --------------
 
